scvi/train/_trainingplans.py

```diff
@@ -141,68 +160,121 @@ def n_obs_training(self, n_obs: int):
         if "n_obs" in self._loss_args:
             self.loss_kwargs.update({"n_obs": n_obs})
         self._n_obs_training = n_obs
+        self.initialize_train_metrics()
+
+    @property
+    def n_obs_validation(self):
+        """
+        Number of observations in the validation set.
+
+        This will update the loss kwargs for loss rescaling.
+
+        Notes
+        -----
+        This can get set after initialization
+        """
+        return self._n_obs_validation
+
+    @n_obs_validation.setter
+    def n_obs_validation(self, n_obs: int):
+        if "n_obs" in self._loss_args:
+            self.loss_kwargs.update({"n_obs": n_obs})
+        self._n_obs_validation = n_obs
+        self.initialize_val_metrics()
 
     def forward(self, *args, **kwargs):
         """Passthrough to `model.forward()`."""
         return self.module(*args, **kwargs)
 
+    def compute_and_log_metrics(
+        self,
+        lossrecorder: LossRecorder,
+        metric_attr_name: str,
+        mode: Literal["train", "validation"],
+    ):
+        """
+        Computes and logs metrics.
+
+        Parameters
+        ----------
+        lossrecorder
+            LossRecorder object from scvi-tools module
+        metric_attr_name
+            The name of the torch metric object to use
+        mode
+            Train or validation, used for logging names
+        """
+        lr = lossrecorder
```

Comment:
potentially unnecessary and confusing to abbrev as lr (learning rate)

Reply:
done